src/braket/circuits/text_diagram_builders/matplotlib_circuit_diagram.py

```diff
@@ -0,0 +1,338 @@
+
+        if all(m.moment_type == MomentType.GLOBAL_PHASE for m in circuit._moments):
+            fig, ax = plt.subplots(figsize=(4, 1))
+            ax.text(0.5, 0.5, f"Global phase: {circuit.global_phase}", ha="center", va="center")
+            ax.axis("off")
+            return fig
+
+        # Get circuit qubits and sort them
+        circuit_qubits = circuit.qubits
+        circuit_qubits.sort()
+
+        # Create figure and axis
+        num_qubits = len(circuit_qubits)
+        fig_width = 10  # Will be adjusted based on circuit size
+        fig_height = max(4, num_qubits * 1.5)
+        fig, ax = plt.subplots(figsize=(fig_width, fig_height))
+
+        # Draw the circuit
+        cls._draw_circuit(ax, circuit, circuit_qubits)
+
+        # Adjust layout and return
+        plt.tight_layout()
+        return fig
+
+    @classmethod
+    def _draw_circuit(cls, ax: Axes, circuit: cir.Circuit, circuit_qubits: QubitSet) -> None:
+        """Draw the circuit on the given axis.
+
+        Args:
+            ax (Axes): Matplotlib axis to draw on
+            circuit (Circuit): Circuit to draw
+            circuit_qubits (QubitSet): Sorted set of qubits in the circuit
+        """
+        # Set up the axis
+        ax.set_aspect("equal")
+        ax.axis("off")
+
+        time_slices = circuit.moments.time_slices()
+        num_columns = len(time_slices) + (1 if circuit.result_types else 0)
+
+        # Draw qubit lines
+        wire_color = getattr(cls, "WIRE_COLOR", "#000000")
+        text_color = getattr(cls, "TEXT_COLOR", "#000000")
+        for i, qubit in enumerate(circuit_qubits):
+            y = -i * cls.WIRE_SPACING
+            ax.plot(
+                [0, num_columns * cls.COLUMN_SPACING],
+                [y, y],
+                color=wire_color,
+                linewidth=1,
+                alpha=1.0,
+            )
+            ax.text(-0.5, y, f"q[{qubit}]", ha="right", va="center", color=text_color)
+
+        # Draw gates and connections
+        x_pos = 0
+
+        for instructions in time_slices.values():
+            # Group instructions to handle overlapping gates
+            groupings = _group_items(circuit_qubits, instructions)
+            for grouping in groupings:
+                cls._draw_instruction_group(ax, grouping[1], circuit_qubits, x_pos)
+                x_pos += cls.COLUMN_SPACING
+
+        # Draw result types if any
+        if circuit.result_types:
+            cls._draw_result_types(ax, circuit.result_types, circuit_qubits, x_pos)
+
+        # Draw global phase if present (after all instructions, so it's not overlapped)
+        if circuit.global_phase != 0:
+            y_center = (
+                0 if len(circuit_qubits) == 1 else -len(circuit_qubits) * cls.WIRE_SPACING / 2
+            )
+            # If global phase is exactly 0.5, display as 0.50π, else as a number with two decimals
+            if abs(circuit.global_phase - 0.5) < 1e-9:
+                phase_str = "0.50π"
```

Comment:
This is cool. Two comments
(1) would be cool to do this for other multiples of `pi`, not just `0.5`
(2) we should share the logic for displaying global phase and for angle parameters to other gates